.travis.yml

```diff
@@ -177,6 +230,55 @@ matrix:
         - mkdir BUILD
       script:
         # Run local mbed 2 testing
-        # Note: These take ~40 minutes to run
-        - python2 -u tools/build_travis.py | sed -n '/^Executing/p'
+        - python2 -u tools/build_travis.py --vendor SILICON_LABS
 
+    - python: '2.7'
+      env:
+        - NAME=mbed2-MAXIM
+      install:
+        # Install dependencies
+        - sudo apt-get install gcc-arm-embedded
+        - pip install --user -r requirements.txt
+        # Print versions we use
+        - arm-none-eabi-gcc --version
+        - python --version
+      before_script:
+        # Create BUILD directory for tests
+        - mkdir BUILD
+      script:
+        # Run local mbed 2 testing
+        - python2 -u tools/build_travis.py --vendor MAXIM
+
+    - python: '2.7'
+      env:
+        - NAME=mbed2-ATMEL
+      install:
+        # Install dependencies
+        - sudo apt-get install gcc-arm-embedded
+        - pip install --user -r requirements.txt
+        # Print versions we use
+        - arm-none-eabi-gcc --version
+        - python --version
+      before_script:
+        # Create BUILD directory for tests
+        - mkdir BUILD
+      script:
+        # Run local mbed 2 testing
+        - python2 -u tools/build_travis.py --vendor ATMEL
+
+    - python: '2.7'
+      env:
+        - NAME=mbed2-NUVOTON
```

Comment:
I bet there's a way to not have to repeat yourself here... but I tried it on my fork and ran into issues.

Will look into this later...